numba_dppy/tests/test_with_context.py

```diff
@@ -16,14 +16,52 @@
 
 import dpctl
 import numpy as np
+import pytest
 from numba import njit
 from numba.tests.support import captured_stdout
 
-import numba_dppy
 import numba_dppy as dppy
 from numba_dppy import config
 
 from . import _helper
+from ._helper import assert_auto_offloading
+
+skip_no_gpu = pytest.mark.skipif(
+    not _helper.has_gpu_queues(), reason="No GPU platforms available"
+)
+skip_no_cpu = pytest.mark.skipif(
+    not _helper.has_cpu_queues(), reason="No CPU platforms available"
+)
+
+
+filter_strings = [
+    pytest.param("level_zero:gpu:0", marks=skip_no_gpu),
+    pytest.param("opencl:gpu:0", marks=skip_no_gpu),
+    pytest.param("opencl:cpu:0", marks=skip_no_cpu),
+]
+
+
+def scenario(filter_str, context):
+    @njit
+    def func(a, b):
+        return a + b
+
+    a = np.ones((64), dtype=np.int32)
+    b = np.ones((64), dtype=np.int32)
+
+    device = dpctl.SyclDevice(filter_str)
+    with context(device):
```

Comment:
How are we testing if `@njit` has actually been offloaded and did not fallback?

Reply:
It is tested in `assert_auto_offloading` call. See below.